cmake/Modules/Packages.cmake

```diff
@@ -186,12 +186,19 @@ if(TOMOPY_USE_CUDA)
 
     if(NOT CMAKE_CUDA_COMPILER AND CUDAToolkit_FOUND)
         set(CMAKE_CUDA_COMPILER ${CUDAToolkit_BIN_DIR}/nvcc)
+        if (WIN32)
+            set(CMAKE_CUDA_COMPILER "${CMAKE_CUDA_COMPILER}.exe")
+        endif (WIN32)
     endif()
 
     enable_language(CUDA)
 
-    list(APPEND EXTERNAL_LIBRARIES "CUDA::nppc_static;CUDA::npps_static;CUDA::nppig_static;CUDA::nppisu_static")
-    list(APPEND EXTERNAL_INCLUDE_DIRS ${CUDA_INCLUDE_DIRS}
+    if (WIN32)
+        list(APPEND TOMOPY_EXTERNAL_LIBRARIES "CUDA::nppc;CUDA::npps;CUDA::nppig;CUDA::nppisu")
+    else()
+        list(APPEND TOMOPY_EXTERNAL_LIBRARIES "CUDA::nppc_static;CUDA::npps_static;CUDA::nppig_static;CUDA::nppisu_static")
+    endif(WIN32)
+    list(APPEND TOMOPY_EXTERNAL_INCLUDE_DIRS ${CUDA_INCLUDE_DIRS}
```

Comment:
```suggestion
    foreach(_NPP_LIB nppc npps nppig nppisu)
        if(TARGET CUDA::${_NPP_LIB}_static)
            list(APPEND TOMOPY_EXTERNAL_LIBRARIES CUDA::${_NPP_LIB}_static)
        elseif(TARGET CUDA::${_NPP_LIB})
            list(APPEND TOMOPY_EXTERNAL_LIBRARIES CUDA::${_NPP_LIB})
        else()
            message(FATAL_ERROR "Missing CUDA NPP target: CUDA::${_NPP_LIB} / CUDA::${_NPP_LIB}_static")
        endif()
    endforeach()
    list(APPEND TOMOPY_EXTERNAL_INCLUDE_DIRS ${CUDA_INCLUDE_DIRS}
```

Using an interface target, if would look like this:

```cmake
# create an interface library
add_library(tomopy-cuda-npp INTERFACE)
# create an alias in the tompy namespace which helps make it clear that you want to link to
# a cmake target named tomopy::cuda-npp, not a potential library tomopy-cuda-npp (i.e. libtomopy-cuda-npp.so) 
add_library(tomopy::cuda-npp ALIAS tomopy-cuda-npp)


if(...)
    ...
    foreach(_NPP_LIB nppc npps nppig nppisu)
        if(TARGET CUDA::${_NPP_LIB}_static)
            target_link_libraries(tomopy-cuda-npp INTERFACE CUDA::${_NPP_LIB}_static)
        elseif(TARGET CUDA::${_NPP_LIB})
            target_link_libraries(tomopy-cuda-npp INTERFACE CUDA::${_NPP_LIB})
        else()
            message(FATAL_ERROR "Missing CUDA NPP target: CUDA::${_NPP_LIB} / CUDA::${_NPP_LIB}_static")
        endif()
    endforeach()
    target_include_directories(tomopy-cuda-npp INTERFACE ${CUDA_INCLUDE_DIRS})
    ...
endif()

# later (even if tomopy::cuda-npp is an empty interface library)
target_link_libraries(libtomopy-accel 
    PUBLIC 
        tomopy::cuda-npp
        ...
    PRIVATE ...)
```

Reply:
Or instead of making an interface library for each external library, you could just do:

```cmake
add_library(tomopy-external INTERFACE)
add_library(tomopy::external ALIAS tomopy-external)

...
            target_link_libraries(tomopy-external INTERFACE CUDA::${_NPP_LIB}_static)
...

target_link_libraries(libtomopy-accel 
    PUBLIC 
        tomopy::external
        ...
    PRIVATE ...)
```

Reply:
Thanks for explaining interface libraries. I was having trouble finding an explanation on the web.

Reply:
Yeah the cmake docs aren't very clear but once you get the hang of them, they are a phenomenal way to bundle optional extensions for compiler flags, include dirs, compile definitions, etc. into something that can be easily added exactly where you want them.